qa/rpc-tests/excessive.py

```diff
@@ -12,17 +12,231 @@
 from test_framework.test_framework import BitcoinTestFramework
 from test_framework.util import assert_equal
 from test_framework.util import *
+from test_framework.blocktools import *
+import test_framework.script as script
 import pdb
+import sys
+import logging
+logging.basicConfig(format='%(asctime)s.%(levelname)s: %(message)s', level=logging.INFO)
 
 class ExcessiveBlockTest (BitcoinTestFramework):
+    def __init__(self,extended=False):
+      self.extended = extended
+      BitcoinTestFramework.__init__(self)
 
-    def run_test (self):
-        BitcoinTestFramework.run_test (self)
+    def setup_network(self, split=False):
+        self.nodes = start_nodes(4, self.options.tmpdir,timewait=60*10)
+        interconnect_nodes(self.nodes)
+        self.is_network_split=False
+        self.sync_all()
+
+        if 0:  # Use to create wallet with lots of addresses
+          TEST_SIZE=10000
+          print "Creating addresses..."
+          addrs = [ self.nodes[0].getnewaddress() for _ in range(TEST_SIZE+1)]
+          with open("walletAddrs.json","w") as f: 
+            f.write(str(addrs))
+            pdb.set_trace()
+
+    def run_test(self):
+      BitcoinTestFramework.run_test (self)
+
+      # exact point in chain is not required
+      #tips = self.nodes[0].getchaintips ()
+      #assert_equal (len (tips), 1)
+      #assert_equal (tips[0]['branchlen'], 0)
+      #assert_equal (tips[0]['height'], 200)
+      #assert_equal (tips[0]['status'], 'active')
+
+      self.testExcessiveBlock()
+      self.testExcessiveTx()
```

Comment:
File length & class/method length are quantitative ones which favors splitting up into smaller modules.
One could argue the excessive.py is getting excessively long :-)